screens/ChatScreen.js

```diff
@@ -42,126 +44,249 @@ class ChatScreen extends Component {
-          user: {
-            _id: m.sender.userId
-          }
+  async getMessage() {
+    this.channel.then((c) => {
+      c.on('messageAdded', message => this.updateLocalMessageStateSingle(message));
+
+      c.getMessages().then((messages) => {
+        const totalMessages = messages.items.length;
+        const localMessages = [];
+
+        for (let i = 0; i < totalMessages; i += 1) {
+          const message = messages.items[i];
+          const currMessage = {
+            _id: `${message.index}`,
+            text: `${message.body}`,
+            createdAt: `${message.timestamp}`,
+            user: {
+              _id: `${message.author}`,
+            },
+          };
+
+          localMessages.push(currMessage);
         }
-      })
+
+        this.setState(previousState => ({
+          messages: previousState.messages.concat(localMessages),
+        }));
+      });
     });
+  }
 
-    this.setState({ sendBirdApp: sendBirdApp });
+  async updateLocalMessageStateSingle(message) {
+    const messages = {
+      items: [message]
+    };
 
-    console.log("Done mapping state"); */
+    this.updateLocalMessageState(messages);
   }
 
-  async onSend(messages = []) {
-    const promises = messages.map(msg => new Promise((resolve) => {
-      this.channel.sendUserMessage(msg.text, resolve);
-    }));
-    await Promise.all(promises);
+  async updateLocalMessageState(messages) {
+    const totalMessages = messages.items.length;
+    const localMessages = [];
+
+    for (let i = 0; i < totalMessages; i += 1) {
+      const message = messages.items[i];
+      const currMessage = {
+        _id: `${message.index}`,
+        text: `${message.body}`,
+        createdAt: `${message.timestamp}`,
+        user: {
+          _id: `${message.author}`,
+        },
+      };
+
+      localMessages.push(currMessage);
+    }
+
     this.setState(previousState => ({
-      messages: previousState.messages.concat(messages),
+      messages: previousState.messages.concat(localMessages),
     }));
   }
 
-  async onReceive(messages) {
-    console.log('onReceive');
-    this.setState(previousState => ({
-      messages: previousState.messages.concat(messages),
-    }));
-    console.log('onReceive done');
+  async createChannelWithUser(chatClient) {
+    console.log('Creating channel...');
+
+    const user = this.props.navigation.state.params.mentee._id;
+    // Sort the IDs so that we have a deterministic order
+    const channelName = this.getChannelName();
+    return chatClient.createChannel({
+      uniqueName: channelName,
+      friendlyName: channelName,
+    }).then((channel) => {
+      console.log('Created channel');
+      channel.join().catch((err) => {
+        console.error(
+          `Couldn't join channel ${channel.friendlyName} because ${err}`
+        );
+      });
+
+      // Invite other user to your channel
+      channel.invite(user).then(() => {
+        console.log('Your friend has been invited!');
```

Comment:
The user wouldn't see this, is this necessary?

Reply:
It's not. Cleaned up